### ISSUE TITLE
Which vc++ runtime version to install for Python 3.3 and cf_freeze 4.3 ?

### ISSUE THREAD

**Issue by anthony-tuininga** [marcelotduarte/cx_Freeze#10]
Originally reported by: **Anonymous**

----------------------------------------

Hi Anthony,

Could you provide info about which VC++ Runtime version has to be installed to run a Python 3.3 Script frozen by cx_freeze 4.3, please?

.exe crashs on startup on missing DLLs...

Thank you,
Regards,
Thorsten

----------------------------------------
- Bitbucket: https://bitbucket.org/anthony_tuininga/cx_freeze/issue/10

**Comment by anthony-tuininga** [comment]
*Original comment by* **Thomas Kluyver (Bitbucket: [takluyver](https://bitbucket.org/takluyver), GitHub: [takluyver](https://github.com/takluyver))**:

----------------------------------------

I would guess it's the 2010 versions, as Python 3.3 is compiled with VS 2010. [x86](http://www.microsoft.com/en-gb/download/details.aspx?id=5555), [x64](http://www.microsoft.com/en-us/download/details.aspx?id=14632).

If that works, we'll add it to the docs.

**Comment by anthony-tuininga** [comment]
*Original comment by* **Anthony Tuininga (Bitbucket: [anthony_tuininga](https://bitbucket.org/anthony_tuininga), GitHub: Unknown)**:

----------------------------------------

Its more than that, unfortunately. Yes, it is 2010 but the Python team changed the startup routines quite a bit as well. I put something together for the beta but things seem to have changed. So I have a bit more work to do on this before Python 3.3 will be supported well. I hope to do that soon -- it is the highest priority thing for me, I believe. Once that is resolved I will be making a new release.

**Comment by anthony-tuininga** [comment]
*Original comment by* **Anthony Tuininga (Bitbucket: [anthony_tuininga](https://bitbucket.org/anthony_tuininga), GitHub: Unknown)**:

----------------------------------------

It has been a while but I have finally found some time to look into this issue further. The release of Python 3.3 changed the contents of the file importlib.h which is a frozen implementation of the importlib machinery required to start up a Python interpreter. I have updated it with the base release version of the file but have also added a Python issue to see whether this is the right approach. Technically I am violating the API by doing so. The issue number is 16467 if you wish to view it.
